### PR TITLE
Add a couple new cases to DateTimeFormat::fix()

### LINKED ISSUE #11938: DateTimeFormat - Failed to parse time string
- [ ] I have searched open and closed issues for duplicates

### Bug Description

Looks like some items have a German time zone string that cannot be parsed.

```
2022-09-26T16:21:34Z worker [WARNING]: DateTimeFormat::convert: exception: DateTime::__construct(): Failed to parse time string (Mo, 19 Sep 2022 14:51:00 +0200) at position 0 (M): The timezone could not be found in the database {"worker_id":"5b6ff7f","worker_cmd":"OnePoll"} - {"file":"DateTimeFormat.php","line":161,"function":"convert","uid":"307ae3","process_id":680430}
2022-09-26T16:21:34Z worker [WARNING]: DateTimeFormat::convert: exception: DateTime::__construct(): Failed to parse time string (Di, 20 Sep 2022 11:12:00 +0200) at position 0 (D): The timezone could not be found in the database {"worker_id":"5b6ff7f","worker_cmd":"OnePoll"} - {"file":"DateTimeFormat.php","line":161,"function":"convert","uid":"307ae3","process_id":680430}
2022-09-26T16:21:34Z worker [WARNING]: DateTimeFormat::convert: exception: DateTime::__construct(): Failed to parse time string (Mi, 21 Sep 2022 07:37:00 +0200) at position 0 (M): The timezone could not be found in the database {"worker_id":"5b6ff7f","worker_cmd":"OnePoll"} - {"file":"DateTimeFormat.php","line":161,"function":"convert","uid":"307ae3","process_id":680430}
2022-09-26T16:21:35Z worker [WARNING]: DateTimeFormat::convert: exception: DateTime::__construct(): Failed to parse time string (Mi, 21 Sep 2022 10:39:00 +0200) at position 0 (M): The timezone could not be found in the database {"worker_id":"5b6ff7f","worker_cmd":"OnePoll"} - {"file":"DateTimeFormat.php","line":161,"function":"convert","uid":"307ae3","process_id":680430}
2022-09-26T16:21:35Z worker [WARNING]: DateTimeFormat::convert: exception: DateTime::__construct(): Failed to parse time string (Mi, 21 Sep 2022 12:24:00 +0200) at position 0 (M): The timezone could not be found in the database {"worker_id":"5b6ff7f","worker_cmd":"OnePoll"} - {"file":"DateTimeFormat.php","line":161,"function":"convert","uid":"307ae3","process_id":680430}
2022-09-26T16:21:35Z worker [WARNING]: DateTimeFormat::convert: exception: DateTime::__construct(): Failed to parse time string (Mi, 21 Sep 2022 12:43:00 +0200) at position 0 (M): The timezone could not be found in the database {"worker_id":"5b6ff7f","worker_cmd":"OnePoll"} - {"file":"DateTimeFormat.php","line":161,"function":"convert","uid":"307ae3","process_id":680430}
2022-09-26T16:21:35Z worker [WARNING]: DateTimeFormat::convert: exception: DateTime::__construct(): Failed to parse time string (Mi, 21 Sep 2022 15:15:00 +0200) at position 0 (M): The timezone could not be found in the database {"worker_id":"5b6ff7f","worker_cmd":"OnePoll"} - {"file":"DateTimeFormat.php","line":161,"function":"convert","uid":"307ae3","process_id":680430}
2022-09-26T16:21:35Z worker [WARNING]: DateTimeFormat::convert: exception: DateTime::__construct(): Failed to parse time string (Sa, 24 Sep 2022 08:37:00 +0200) at position 0 (S): The timezone could not be found in the database {"worker_id":"5b6ff7f","worker_cmd":"OnePoll"} - {"file":"DateTimeFormat.php","line":161,"function":"convert","uid":"307ae3","process_id":680430}
```

Not sure if this is an issue with the source format or with our datetime parser.

### Platform Info

Friendica Version:
2022.09-rc

Friendica Source: 
git

PHP version:
7.4

SQL version:
MariaDB

### PR DESCRIPTION
**Pull request by MrPetovan**
Re-fix #11938

- Reworked method to perform more string replacements and fewer regular expression matches

### CODE REVIEW

File: src/Util/DateTimeFormat.php [hunk 1]
```diff
@@ -184,17 +184,17 @@ public static function convert($s = 'now', $tz_to = 'UTC', $tz_from = 'UTC', $fo
 	 */
 	public static function fix(string $dateString): string
 	{
-		$patterns = [
-			['#(\w+), (\d+/\d+/\d+) - (\d+:\d+)#', '$1, $2 $3'],
-			['#(\d+-\d+-\d+)T(\d+:\d+:\d+)ZZ#', '$1T$2Z'],
-			['#(\d+-\d+-\d+)T(\d+:\d+:\d+\.\d+)ZZ#', '$1T$2Z'],
+		$search  = ['Mär', 'März', 'Mai', 'Juni', 'Juli', 'Okt', 'Dez', 'ET' , 'ZZ', ' - ', '&#x2B;'];
```
Comment (annando) [inline src/Util/DateTimeFormat.php:185]: We mustn't use these simple patterns to fix the problem. We are not only calling this function with fixed date values but also with date formulas. When we replace " - " with "" then a date formula like "now - 3 days" will stop working. I accidentally deleted many posts this way some weeks ago :grin:

**Review by annando — CHANGES_REQUESTED** [review]

File: src/Util/DateTimeFormat.php (hunk 1 above)
Comment (MrPetovan) [reply src/Util/DateTimeFormat.php:185]: This is where function comments are helping in a big way. As long as this kind of knowledge stays in your head, it can only help yourself. As soon as you put it down for others, it helps everybody save their time.

**Comment by MrPetovan** [comment]
@annando I have added the relative date string to the test dataset, tweaked the method and we're good, unless you know about another corner case that we should add to the tests as well?

File: src/Util/DateTimeFormat.php (hunk 1 above)
Comment (nupplaphil) [reply src/Util/DateTimeFormat.php:185]: And adding each edge case a test so we can avoid such regressions ;-)))

**Comment by annando** [comment]
I only stumbled across this ` - ` thing. But I just had an idea. What if we only apply the fix when the first call of `$d = new DateTime($s, $from_obj);` fails? Doing so, we will avoid any side effects, I guess.

File: src/Util/DateTimeFormat.php (hunk 1 above)
Comment (annando) [reply src/Util/DateTimeFormat.php:185]: Problem is that it is not really an edge case. It is based on the way that functionality works. This functionality is called both with static values and with formulas. And thus problems like these can always occur.

**Comment by MrPetovan** [comment]
This is a good idea.

**Comment by MrPetovan** [comment]
This feels slightly bad to me but I'm glad we saved hundreds of unwarranted calls to `DateTimeFormat::fix()`. Still, I've added a canary test to make sure that relative date/time strings still work as intended given the mangling risk.

**Review by nupplaphil — APPROVED** [review]

**Review by annando — APPROVED** [review]
